File: backend-java/src/main/java/com/backend/java/backendjava/infra/entities/Card.java
package com.backend.java.backendjava.infra.entities;

import com.backend.java.backendjava.dtos.requests.CardRequest;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "card_tb")
public class Card implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String userDocument;
    @Column
    private String creditCardToken;
    @Column
    private Long cardValue;

    public Card(CardRequest request) {
        this.userDocument = request.userDocument();
        this.creditCardToken = request.creditCardToken();
        this.cardValue = request.cardValue();
    }

    public void updateCard(Card card) {
        userDocument = card.getUserDocument();
        creditCardToken = card.getCreditCardToken();
        cardValue = card.getCardValue();
    }
}

File: backend-kotlin/src/main/kotlin/com/backend/kotlin/backendkotlin/infra/entities/Card.kt
package com.backend.kotlin.backendkotlin.infra.entities

import com.backend.kotlin.backendkotlin.dtos.requests.CardRequest
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table


@Entity
@Table(name = "card_tb")
data class Card(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    @Column(nullable = false, unique = true)
    var userDocument: String,
    @Column(nullable = false, unique = true)
    var creditCardToken: String,
    @Column(nullable = false)
    var cardValue: Long
) {
    constructor(request: CardRequest) : this(null, request.userDocument, request.creditCardToken, request.cardValue)

    fun updateCard(card: Card) {
        userDocument = card.userDocument
        creditCardToken = card.creditCardToken
        cardValue = card.cardValue
    }
}
